lib/bundler/inject/dsl_patch.rb

```diff
@@ -97,6 +99,10 @@ def load_bundler_d(dir)
           eval_gemfile(f)
         end
       end
+
+      def search_path(calling_file)
+        [File.dirname(calling_file)] + ENV.fetch("BUNDLE_BUNDLER_INJECT__GEM_PATH", "").split(':')
```

Comment:
We probably should specify in the README what the search order is, I think this is perfect though prioritize the local dir then in order in the BUNDLE_BUNDLER_INJECT__GEM_PATH

Reply:
Minor, but let's move any env var handling to the same place that skip_warnings? is, and wrap it in a proper method.

Reply:
Also, once you move over there, you'll see the preferable way to access the settings is with `Bundler.settings["bundler_inject.gem_path"]`.  This allows all of ENV var, .bundle/config file, global bundler config, and local bundler config.